lib/modules/manager/poetry/schema.ts

```diff
@@ -132,58 +134,151 @@ export const PoetryDependencyRecord = LooseRecord(
     dep.skipReason = 'invalid-version';
     return dep;
   })
-).transform((record) => {
-  for (const [depName, dep] of Object.entries(record)) {
-    dep.depName = depName;
-    if (!dep.packageName) {
-      const pep503NormalizeRegex = regEx(/[-_.]+/g);
-      const packageName = depName
-        .toLowerCase()
-        .replace(pep503NormalizeRegex, '-');
-      if (depName !== packageName) {
-        dep.packageName = packageName;
+)
+  .transform((record) => {
+    const deps: PackageDependency[] = [];
+    for (const [depName, dep] of Object.entries(record)) {
+      dep.depName = depName;
+      if (!dep.packageName) {
+        const pep503NormalizeRegex = regEx(/[-_.]+/g);
+        const packageName = depName
+          .toLowerCase()
+          .replace(pep503NormalizeRegex, '-');
+        if (depName !== packageName) {
+          dep.packageName = packageName;
+        }
       }
+      deps.push(dep);
     }
-  }
-  return record;
-});
-
+    return deps;
+  })
+  .catch([]);
 export type PoetryDependencyRecord = z.infer<typeof PoetryDependencyRecord>;
 
+function withDepType<
+  Output extends PackageDependency[],
+  Schema extends ZodType<Output, ZodTypeDef, unknown>
+>(schema: Schema, depType: string): ZodEffects<Schema> {
+  return schema.transform((deps) => {
+    for (const dep of deps) {
+      dep.depType = depType;
+    }
+    return deps;
+  });
+}
+
 export const PoetryGroupRecord = LooseRecord(
   z.string(),
-  z.object({
-    dependencies: PoetryDependencyRecord.optional(),
+  z
+    .object({ dependencies: PoetryDependencyRecord })
+    .transform(({ dependencies }) => dependencies)
+)
+  .transform((record) => {
+    const deps: PackageDependency[] = [];
+    for (const [groupName, group] of Object.entries(record)) {
+      for (const dep of Object.values(group)) {
+        dep.depType = groupName;
+        deps.push(dep);
+      }
+    }
+    return deps;
   })
-);
+  .catch([]);
```

Comment:
same here. what can go wrong?

Reply:
These schemas are being used in the optional fields and we're merging them all into the single array. Unlike objects, array spread operator doesn't work well with undefined values, so I think the better course of action is to fallback to `[]`.